src/D2L.Security.OAuth2.WebApi/Authorization/OAuth2AuthorizeAttribute.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Linq;
+using System.Web.Http;
+using System.Web.Http.Controllers;
+
+namespace D2L.Security.OAuth2.Authorization {
+	public abstract class OAuth2AuthorizeAttribute : AuthorizeAttribute {
+
+		private const string AUTH_HAS_RUN = "D2L.Security.OAuth2.Authorization.OAuth2AuthorizeAttribute_Auth_Has_Run";
+		private const string FAILING_ATTR_PROP = "D2L.Security.OAuth2.Authorization.OAuth2AuthorizeAttribute_Failing_Attribute";
+
+		protected override bool IsAuthorized( HttpActionContext actionContext ) {
+			var props = actionContext.Request.Properties;
+			if( props.ContainsKey( AUTH_HAS_RUN ) ) {
+				return true;
+			}
+
+			props.Add( AUTH_HAS_RUN, true );

```

Comment:
nit: nice if literal args have explicit arg names so I don't have to find docs ;) (lazy)